### PR TITLE
chore: swap changelog generation to generate-changelog

### PR DESCRIPTION
**Pull request by mkilpatrick**
This swaps from using `conventional-changelog-cli` to `generate-changelog` since the former had issues with rebases.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+scripts/
```
Comment (mancusi) [inline .eslintignore:1]: I think we should continue to lint all our JS/TS even in the scripts directory.

File: scripts/release.ts [hunk 2]
```diff
@@ -81,16 +82,13 @@ step("\nUpdating package version...");
...
-  "-p",
-  "angular",
-  "-i",
-  "CHANGELOG.md",
-  "-s",
-  "--commit-path",
-  ".",
-];
+const latestTag = await getLatestTag(pkgName);
+if (!latestTag) process.exit();
```
Comment (mancusi) [inline scripts/release.ts:86]: wrap with curly braces

File: scripts/release.ts [hunk 3]
```diff
@@ -81,16 +82,13 @@ step("\nUpdating package version...");
...
+const sha = await run("git", ["rev-list", "-n", "1", latestTag], {
+  stdio: "pipe",
+}).then((res) => res.stdout.trim());
```
Comment (mancusi) [inline scripts/release.ts:89]: if you're awaiting then you don't need to use a `.then`

**Review by mancusi — CHANGES_REQUESTED** [review]

File: scripts/release.ts (hunk 3 above)
Comment (mkilpatrick) [reply scripts/release.ts:91]: Need to do this since the await returns ExecReturnValue<string>

File: scripts/releaseUtils.ts [hunk 4]
```diff
@@ -100,7 +100,7 @@ export function getVersionChoices(currentVersion: string): VersionChoice[] {
   const isStable = !currentBeta && !currentAlpha;
 
   function inc(i: ReleaseType, tag = currentAlpha ? "alpha" : "beta") {
-    return semver.inc(currentVersion, i, tag)!;
+    return semver.inc(currentVersion, i, tag) || "0";
```
Comment (mancusi) [inline scripts/releaseUtils.ts:103]: what was the reason for this change?

We definitely want to throw an error if this lib doesn't return a result we'd expect.

**Review by mancusi — CHANGES_REQUESTED** [review]

File: scripts/releaseUtils.ts (hunk 4 above)
Comment (mkilpatrick) [reply scripts/releaseUtils.ts]: lint warning - updated to throw an error

**Review by mancusi — APPROVED** [review]

**Pull request merged**